packages/message-manager/src/AbstractMessageManager.ts

```diff
@@ -127,6 +146,52 @@ export abstract class AbstractMessageManager<
 
   private readonly additionalFinishStatuses: string[];
 
+  /**
+   * Adds request props to the messsage params and returns a new messageParams object.
+   * @param messageParams - The messageParams to add the request props to.
+   * @param req - The original request object.
+   * @returns The messageParams with the request props added.
+   */
+  protected addRequestToMessageParams(
```

Comment:
Rather than making this optional and an additional protected method, could we just encapsulate the logic within `createUnapprovedMessage` so it's automatic?

Reply:
One thing is I will need to update messageParams here instead of creating a new object because it is later emitted in the addUnapprovedMessage methods

Reply:
There are two reasons I'm leaning towards not calling addRequestToMessageParams in createUnapprovedMessage: 
- we would need to update messageParams here instead of creating a new object because it is later emitted in the addUnapprovedMessage methods. This would no longer make this function a pure function
- messageParams may be updated within addUnapprovedMessage themselves too. I think it's better to have these updates live together rather than in two different locations

Reply:
I was forgetting that the params and message itself are distinct things!